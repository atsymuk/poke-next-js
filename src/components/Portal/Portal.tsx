import { ReactNode, useEffect, useState, VFC } from 'react'
import ReactDOM from 'react-dom'

interface PortalProps {
  children: ReactNode
}

const Portal: VFC<PortalProps> = ({ children }) => {
  const [container] = useState(() => {
    if (typeof window === 'undefined') {
      return
    }

    return document.createElement('div')
  })

  useEffect(() => {
    if (!container) {
      return
    }

    document.body.appendChild(container)
    return () => {
      document.body.removeChild(container)
    }
  }, [container])

  if (!container) {
    return null
  }

  return ReactDOM.createPortal(children, container)
}

export default Portal
