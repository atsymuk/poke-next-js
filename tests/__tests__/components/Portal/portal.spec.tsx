import { render } from '@testing-library/react'
import Portal from 'components/Portal/Portal'

it('Portal component snapshot', () => {
  const { container } = render(
    <Portal>
      <></>
    </Portal>,
  )
  expect(container).toMatchSnapshot()
})
