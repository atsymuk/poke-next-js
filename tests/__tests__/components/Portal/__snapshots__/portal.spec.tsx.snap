// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal component snapshot 1`] = `<div />`;
